core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -128,6 +131,15 @@ private void validateInputFields(Schema inputSchema) {
     } else if ((inputSchema.getField(conf.getFieldValue())) == null) {
       throw new IllegalArgumentException(
         String.format("Fieldvalue '%s' does not exist in input schema %s", conf.getFieldValue(), inputSchema));
+    } else if (!inputSchema.getField(conf.getKeyField()).getSchema().getType().equals(Schema.Type.STRING)) {
+      throw new IllegalArgumentException(
+        String.format("Keyfield '%s' must be of type String", conf.getKeyField()));
+    } else if (!inputSchema.getField(conf.getFieldName()).getSchema().getType().equals(Schema.Type.STRING)) {
+      throw new IllegalArgumentException(
+        String.format("Fieldname '%s' must be of type String", conf.getFieldName()));
+    } else if (!inputSchema.getField(conf.getFieldValue()).getSchema().getType().equals(Schema.Type.STRING)) {
```

Comment:
I believe this check is done for the input parameters (key, fieldvalue and fieldname). The decision of having String type is for the output that we are going to emit.
Here we are trying to ensure that input that plug-in is receiving is having String type.